authority/provisioner/oidc.go

```diff
@@ -377,6 +393,11 @@ func (o *OIDC) AuthorizeSSHSign(ctx context.Context, token string) ([]SignOption
 	if err != nil {
 		return nil, errs.Wrap(http.StatusInternalServerError, err, "oidc.AuthorizeSSHSign")
 	}
+	// Reuse the contains function provided for simplicity
+	if !containsAllMembers(iden.Usernames, []string{claims.Username}){
+		// Add preferred_username to the identity's Username
+		iden.Usernames = append(iden.Usernames, claims.Username)
+	}
```

Comment:
We should change the `GetIdentityFunc` to pass both email and PreferrendUsername.
https://github.com/smallstep/certificates/blob/582d6b161d1e0744c3f0d152b0a26734cd127b42/authority/provisioner/provisioner.go#L340-L367

We want to be able to get all usernames with that one.

Reply:
And we should not add it if it's empty.

Reply:
I'm still new to golang, but would something like this work?
I'm not sure what logic you would prefer for removing duplicates, so is it ok to leave it to you to choose that?

I've added write privileges to the smallstep team so you can go ahead and edit in my off timezone